version: "3.9"
services:
  arch:
    image: marcpartensky/arch:vnc
    restart: unless-stopped
    ports: # vnc is shared
      - "5900:5900"
    networks:
      - caddy
      - ldap
    volumes:
      - arch:/home/marc
    labels:
      caddy: arch.marcpartensky.com
      caddy.reverse_proxy: "{{upstreams 8080}}"
      traefik.enable: "true"
      traefik.http.routers.arch.rule: Host(`arch.marcpartensky.com`)
      traefik.http.routers.arch.entrypoints: websecure
      traefik.http.routers.arch.tls.certresolver: tls
      traefik.http.services.arch.loadbalancer.server.port: 8080
      traefik.http.routers.arch.middlewares: ldap_auth
      traefik.http.middlewares.ldap_auth.plugin.ldapAuth.enabled: "true"
      traefik.http.middlewares.ldap_auth.plugin.ldapAuth.logLevel: "DEBUG"
      traefik.http.middlewares.ldap_auth.plugin.ldapAuth.url: ldaps://ldap.marcpartensky.com
      traefik.http.middlewares.ldap_auth.plugin.ldapAuth.port: 636
      traefik.http.middlewares.ldap_auth.plugin.ldapAuth.useTLS: "true"
      traefik.http.middlewares.ldap_auth.plugin.ldapAuth.insecureSkipVerify: "true"
      traefik.http.middlewares.ldap_auth.plugin.ldapAuth.baseDN: "ou=people,dc=admin,dc=org"
      traefik.http.middlewares.ldap_auth.plugin.ldapAuth.attribute: uid

volumes:
  arch:

networks:
  ldap:
    external: true
  caddy:
    external: true
